projects/com.oracle.truffle.llvm.nodes.impl/src/com/oracle/truffle/llvm/nodes/impl/memory/load/LLVMDirectLoadNode.java

```diff
@@ -91,6 +93,22 @@ public LLVMFunctionDescriptor executeAddress(LLVMAddress addr) {
         public LLVMAddress executeAddress(LLVMAddress addr) {
             return LLVMMemory.getAddress(addr);
         }
+
+        // The following specializations are simpler than supporting arbitrary foreign objects here
+
+        @Specialization
+        public Object executeAddress(ManagedMallocObject addr) {
+            return addr.get(0);
+        }
+
+        @Specialization(guards = "objectIsManagedMalloc(addr)")
```

Comment:
Can this specialization occur? Isn't it already covered by the first specialization?